### PR TITLE
Destination Snowflake:  Storage ops to support refreshes

### PR DESCRIPTION
**Pull request by gisripa**
## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by gisripa** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/39473?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#39473** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/39473?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#38622** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38622?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#38616](https://github.com/airbytehq/airbyte/pull/38616) <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38616?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#38713](https://github.com/airbytehq/airbyte/pull/38713) <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38713?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#38067** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38067?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#38086](https://github.com/airbytehq/airbyte/pull/38086) <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38086?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#38331** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38331?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#38075** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38075?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#38121** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38121?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#39107](https://github.com/airbytehq/airbyte/pull/39107) <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/39107?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#38770** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38770?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gisripa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseTypingDedupingTest.kt
Comment (edgao) [inline airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseTypingDedupingTest.kt:235]: nit: rename to catalog1 for clarity

File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseTypingDedupingTest.kt
Comment (edgao) [inline airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseTypingDedupingTest.kt:244]: let's make these use APPEND, since we want to eventually kill overwrite

File: airbyte-integrations/connectors/destination-snowflake/src/test-integration/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeInternalStagingLowercaseDatabaseTypingDedupingTest.kt
Comment (edgao) [inline airbyte-integrations/connectors/destination-snowflake/src/test-integration/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeInternalStagingLowercaseDatabaseTypingDedupingTest.kt:43]: your call: want to remove these? (I don't care either way, but if we keep them it's probably worth adding a comment like `overridden so that we can run directly from intellij`)

**Review by edgao — APPROVED** [review]
:fire: some nitpicks/clarifications, LGTM after resolving those :shipit:

File: airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.kt
Comment (stephane-airbyte) [reply airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.kt:105]: no can do. That's a java function (ResultSet.getTables)

**Pull request merged**
